src/Symfony/Bundle/FrameworkBundle/Command/CachePoolPruneCommand.php

```diff
@@ -0,0 +1,121 @@
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Cache\PruneableInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\Argument\RewindableGenerator;
+
+/**
+ * Cache pool pruner command.
+ */
+final class CachePoolPruneCommand extends Command
+{
+    /**
+     * @var RewindableGenerator
+     */
+    private $pools;
+
+    /**
+     * @param RewindableGenerator $pools
+     */
+    public function __construct(RewindableGenerator $pools)
+    {
+        parent::__construct();
+        $this->pools = $pools;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('cache:pool:prune')
+            ->addArgument('pools', InputArgument::IS_ARRAY | InputArgument::OPTIONAL, 'A list of cache pool service ids')
+            ->setDescription('Prune cache pools')
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> command prunes all or the given cache pools.
+
+    %command.full_name% [<cache pool service id 1> [...<cache pool service id N>]]
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        if (0 === count($pools = $this->getMatchingPools($input))) {
+            throw new RuntimeException('No pruneable cache pools found.');
+        }
+
+        foreach ($pools as $name => $pool) {
+            $io->comment(sprintf('Pruning cache pool: <info>%s</info>', $name));
+            $pool->prune();
+        }
+
+        $io->success('Cache was successfully pruned.');
+    }
+
+    /**
+     * @param InputInterface $input
+     *
+     * @return PruneableInterface[]
+     */
+    private function getMatchingPools(InputInterface $input)
+    {
+        $allPools = $this->getPrunablePools();
+        $reqNames = $input->getArgument('pools');
+
+        if (0 === count($reqNames)) {
+            return $allPools;
+        }
+
+        $reqPools = array();
+        foreach ($input->getArgument('pools') as $name) {
+            if (!isset($allPools[$name])) {
+                throw new InvalidArgumentException(sprintf('The "%s" pool does not exist or is not pruneable.', $name));
+            }
+
+            $reqPools[$name] = $allPools[$name];
+        }
+
+        return $reqPools;
+    }
+
+    /**
+     * @return PruneableInterface[]
+     */
+    private function getPrunablePools()
+    {
+        $pools = array();
+
+        foreach ($this->pools as $name => $pool) {
```

Comment:
the issue of using an iterator is that this instantiate the skipped pools. 
so if we want to fine tune, using a service locator would prevent that (then we'd need an additional arg to the constructor: the list of pools found in the locator)